semgrep-core/src/parsing/Parse_rule.ml

```diff
@@ -210,200 +236,191 @@ let parse_options json =
 (* Sub parsers patterns and formulas *)
 (*****************************************************************************)
 
-type _env = string * R.xlang
-
-let parse_pattern (id, lang) s =
-  match lang with
-  | R.L (lang, _) -> R.mk_xpat (Sem (H.parse_pattern ~id ~lang s, lang)) s
+let parse_pattern env e =
+  let s =
+    match e with
+    | G.L (String (value, _)) -> value
+    | G.N (Id ((value, _), _)) -> value
+    | _ -> error ("Expected a string value for " ^ env.id)
+  in
+  let start, end_ = Visitor_AST.range_of_any (G.E e) in
```

Comment:
This is for later, but note that `start` and `end_` are currently the same (each pattern is only associated with one token). This might be really annoying to change (we need to compute an accurate `end_`, but the string given to us by the yaml parser has tabs removed).